drivers/nrf24l01p_ng/nrf24l01p_ng_netdev.c

```diff
@@ -0,0 +1,663 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @ingroup drivers_nrf24l01p_ng
+ * @{
+ *
+ * @file
+ * @brief   Implementation of RIOT's netdev_driver API
+ *          for the NRF24L01+ (NG) transceiver
+ *
+ * @author Fabian Hüßler <fabian.huessler@ovgu.de>
+ * @}
+ */
+#include <errno.h>
+#include <string.h>
+#include <assert.h>
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#include "kernel_defines.h"
+#include "iolist.h"
+#include "irq.h"
+#include "luid.h"
+#include "mutex.h"
+#include "net/eui64.h"
+#include "net/netdev.h"
+#include "xtimer.h"
+
+#include "gnrc_netif_nrf24l01p_ng.h"
+#include "nrf24l01p_ng_constants.h"
+#include "nrf24l01p_ng_registers.h"
+#include "nrf24l01p_ng_communication.h"
+#include "nrf24l01p_ng_states.h"
+#include "nrf24l01p_ng_netdev.h"
+#if IS_USED(MODULE_NRF24L01P_NG_DIAGNOSTICS)
+#include "nrf24l01p_ng_diagnostics.h"
+#endif
+
+#define NRF24L01P_NG_FLG_IRQ \
+    (NRF24L01P_NG_FLG_MAX_RT | NRF24L01P_NG_FLG_TX_DS | NRF24L01P_NG_FLG_RX_DR)
+
+static
+int _init(netdev_t *netdev);
+static
+int _recv(netdev_t *netdev, void *buf, size_t len, void *info);
+static
+int _send(netdev_t *netdev, const iolist_t *iolist);
+static
+void _isr(netdev_t *netdev);
+static
+int _get(netdev_t *netdev, netopt_t opt, void *val, size_t max_len);
+static
+int _set(netdev_t *netdev, netopt_t opt, const void *val, size_t max_len);
```

Comment:
That's a bit of an odd style

Reply:
Usually I would not write it like that, but Murdock already complains if lines are longer than 80 characters. That was often the case for `static` / `static inline`, so I unified the function headers to look like that.
I personally like it more, if the list of arguments is not broken into multiple lines:
```C
static int _set(netdev_t *netdev, netopt_t opt, const void *val,
                size_t max_len);
```
But if you prefer, I´ll change it

Reply:
80 characters is only a suggestion, a lot of code doesn't follow that rule